### PR TITLE
Remove build artifacts

### PR DESCRIPTION
**Pull request by MeanBoyCousin**
# Task: Clean out build artifacts being checked in.

## Description

- Removes unused build artifacts and types from the front-end package.
- Stops the copying of artifacts over to the front-end package.
- Only copies over required types to the front-end package.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update
- [ ] Quality of life improvement

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have NATSPEC'd any new functions
- [ ] If appropriate, I have properly tested my new functionality

### CODE REVIEW

File: packages/contracts/package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
 	},
 	"scripts": {
 		"buyoption:testnet": "npx hardhat run scripts/buySample.ts --network arbitrumRinkeby",
-		"compile": "hardhat clean && hardhat compile && hardhat typechain && ./copyABIsToFrontend.sh",
+		"compile": "hardhat clean && hardhat compile && ./copyABIsToFrontend.sh",
```
Comment (kjr217) [inline packages/contracts/package.json:20]: Have you tried running this without hardhat typechain?

Also for Beyond we will need to add more contract abis to the frontend

Comment (MeanBoyCousin) [reply packages/contracts/package.json:20]: Yes, runs fine. I noticed in their docs that it mentioned you no longer need to do this in newer versions as it happens automatically if it is imported in your hardhat config.

If we need to add more types to the front end, we can just update the shell script as we go whenever the team needs to grab one of the type files generated by the compiler.

Comment (kjr217) [reply packages/contracts/package.json:20]: Lgtm

Comment (MeanBoyCousin) [reply packages/contracts/package.json:20]: Great. Would like @devey to comment as they initially raised the issues but good to go if they agree.

File: packages/contracts/copyABIsToFrontend.sh [hunk 2]
```diff
@@ -1,5 +1,12 @@
-rm -r ../front-end/src/artifacts
-rm -r ../front-end/src/types
+# Clean out the types directory.
```
Comment (devey) [inline packages/contracts/copyABIsToFrontend.sh:1]: we can also create an issue for copying over the _ABIs_ that are relevant to front-end that are currently being hand copied

Comment (MeanBoyCousin) [reply packages/contracts/copyABIsToFrontend.sh:1]: Yes makes sense. When do they get generated?

Comment (devey) [reply packages/contracts/copyABIsToFrontend.sh:1]: @MeanBoyCousin under `packages/contracts/artifacts/contracts`

Comment (MeanBoyCousin) [reply packages/contracts/copyABIsToFrontend.sh:1]: Ok cool. Should be able to use `jq` to do this! Will add it in.

Comment (MeanBoyCousin) [reply packages/contracts/copyABIsToFrontend.sh:6]: @kjr217 @devey

Have added some scripting using `jq` to copy over ABIs. Please can I get a final review?

File: packages/contracts/copyABIsToFrontend.sh [hunk 3]
```diff
@@ -9,4 +14,30 @@ cp \
...
\ No newline at end of file
+../front-end/src/types/
+
+# Clean out the ABIs directory.
+rm -rf ../front-end/src/abis/*
+
+### ABI ###
+
+# Copy only the required ABIs from artifacts dir.
+for FILE_NAME in 'LiquidityPool' 'OptionHandler' 'OptionRegistry' 'PortfolioValuesFeed' 'PriceFeed'
```
Comment (kjr217) [inline packages/contracts/copyABIsToFrontend.sh:25]: OptionHandler doesnt exist as a contract anymore @devey what is this supposed to be? The correct contract is AlphaOptionHandler for your usecase.

PortfolioValuesFeed is incorrect too, the correct contract is AlphaPortfolioValuesFeed.

Comment (devey) [reply packages/contracts/copyABIsToFrontend.sh:25]: @kjr217 I think the ones we are using are actually `Alpha` but were copied without being renamed.

@MeanBoyCousin you can copy over the alpha versions and update the imports just so we don't get confused again in the future

Comment (MeanBoyCousin) [reply packages/contracts/copyABIsToFrontend.sh]: Ok, that should be good for review now.

File: packages/contracts/copyABIsToFrontend.sh [hunk 4]
```diff
@@ -1,5 +1,43 @@
...
+# Clean out the types directory.
+rm -rf ../front-end/src/types/*
+
+# Copy only the required types.
+cp \
+./types/common.ts \
+./types/ERC20.ts \
+./types/LiquidityPool.ts \
+./types/OptionRegistry.ts \
+./types/PortfolioValuesFeed.ts \
```
Comment (kjr217) [inline packages/contracts/copyABIsToFrontend.sh:15]: this is wrong

**Pull request merged**